zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/AnonymousAuthorizationTest.java

```diff
@@ -0,0 +1,184 @@
+
+    // when
+    final var processInstanceKey =
+        ENGINE.processInstance().ofBpmnProcessId(PROCESS_ID).create(anonymous);
+
+    // then
+    assertThat(
+            RecordingExporter.processInstanceRecords()
+                .withProcessInstanceKey(processInstanceKey)
+                .exists())
+        .isTrue();
+  }
+
+  @Test
+  public void shouldNotCreateProcessInstanceWhenDisablingAnonymousClaim() {
+    // given
+    final var anonymousWithExplicitlyDisabledAnonymous =
+        AuthorizationUtil.getAuthInfoWithClaim(Authorization.AUTHORIZED_ANONYMOUS_USER, false);
+
+    // when
+    ENGINE
+        .processInstance()
+        .ofBpmnProcessId(PROCESS_ID)
+        .expectRejection()
+        .create(anonymousWithExplicitlyDisabledAnonymous);
+
+    // then
+    assertThat(RecordingExporter.records().withRejectionType(RejectionType.UNAUTHORIZED).exists())
+        .isTrue();
+  }
+
+  @Test
+  public void shouldNotCreateProcessInstanceWithoutAnonymousClaim() {
+    // given
+    final var anonymousWithExplicitlyDisabledAnonymous =
+        AuthorizationUtil.getAuthInfoWithClaim("foo", "bar");
+
+    // when
+    ENGINE
+        .processInstance()
+        .ofBpmnProcessId(PROCESS_ID)
+        .expectRejection()
+        .create(anonymousWithExplicitlyDisabledAnonymous);
+
+    // then
+    assertThat(RecordingExporter.records().withRejectionType(RejectionType.UNAUTHORIZED).exists())
+        .isTrue();
+  }
+
+  @Test
+  public void shouldResolveWithAnonymousClaim() {
+    // given
+    ENGINE
+        .deployment()
+        .withXmlResource(PROCESS_WITH_NOT_EXISTING_FORM)
+        .withTenantId(TENANT)
+        .deploy(userKey);
+
+    final long processInstanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID_WITH_NOT_EXISTING_FORM)
+            .withTenantId(TENANT)
+            .create();
+
+    // assume
+    final var incident =
+        RecordingExporter.incidentRecords(IncidentIntent.CREATED)
+            .withProcessInstanceKey(processInstanceKey)
+            .getFirst();
+    assertThat(incident).isNotNull();
+
+    // create anonymous authorization
+    final var anonymous =
+        AuthorizationUtil.getAuthInfoWithClaim(Authorization.AUTHORIZED_ANONYMOUS_USER, true);
+
+    // when
+    // deploy missing form
+    ENGINE.deployment().withJsonClasspathResource(TEST_FORM_1).withTenantId(TENANT).deploy();
+    // resolve incident by using anonymous authorization
+    ENGINE.incident().ofInstance(processInstanceKey).withKey(incident.getKey()).resolve(anonymous);
+
+    // then
+    assertThat(RecordingExporter.incidentRecords().onlyEvents())
+        .extracting(io.camunda.zeebe.protocol.record.Record::getKey, Record::getIntent)
+        .describedAs("form not found incident is resolved and no new incident is created")
+        .containsExactly(
+            tuple(incident.getKey(), IncidentIntent.CREATED),
+            tuple(incident.getKey(), IncidentIntent.RESOLVED));
```

Comment:
```suggestion
    assertThat(RecordingExporter.incidentRecords().onlyEvents().limit(2))
        .extracting(io.camunda.zeebe.protocol.record.Record::getKey, Record::getIntent)
        .describedAs("form not found incident is resolved and no new incident is created")
        .containsExactly(
            tuple(incident.getKey(), IncidentIntent.CREATED),
            tuple(incident.getKey(), IncidentIntent.RESOLVED));
```

❌ We must add a limit here otherwise the test will always run 5+ seconds.

Reply:
Applied the same here: https://github.com/camunda/camunda/pull/25741/files#diff-f85f73dd956ccd6f1df51a7128962c2c68479b12c21daae2a7b68618a6a60300R199 - Because that's where I copied the test from and adjusted it.